Add helper function for creating Review Club <details> lists

harding opened the pull request:
Closes #385 

This provides an `{% include %}` template that can be used to create the `<details>` lists currently used for the monthly Bitcoin Core PR Review Club summaries, although it's also hopefully generic enough that it can be reused for any other `<details>` lists we create.  Here's a *before* screenshot of a manually created list from https://bitcoinops.org/en/newsletters/2020/03/11/#bitcoin-core-pr-review-club with one item expanded:

![2020-04-08-11_55_24_873098868](https://user-images.githubusercontent.com/61096/78805800-e3f52700-798f-11ea-9a92-0c361984cfd2.png)

Here's the same list using this helper:

![2020-04-08-12_11_02_601394880](https://user-images.githubusercontent.com/61096/78807439-13a52e80-7992-11ea-9150-93f713c081c9.png)

There's some extra whitespace in the second example; that wasn't deliberate but I liked it so I haven't removed it---but I can if anyone wants it gone.

---

If you build the website using `JEKYLL_ENV=email`, e.g. `JEKYLL_ENV=email make clean preview`, the list is formatted the same way @bitschmidty has used for the emailed newsletter.  E.g., here's a screenshot of Newsletter 88:

![2020-04-08-12_02_07_401103347](https://user-images.githubusercontent.com/61096/78806565-d7bd9980-7990-11ea-8a90-871d1ccd5acc.png)


And here's the `env=email` version:

![2020-04-08-11_46_36_128080776](https://user-images.githubusercontent.com/61096/78806423-9f1dc000-7990-11ea-82c3-453fa1c6c070.png)

---

Feedback requested, particularly from @jnewbery, @jonatack, and @bitschmidty as those most likely to use this, about whether it looks practical.  If it seems good, the todos are:

- [x] Convert the remaining `<details>` lists in Newsletters 88 and 92 (both English and Japanese) to use this helper
- [x] Remove the old `review-club-questions` CSS class

bitschmidty commented on assets/css/main.scss:
```diff
@@ -265,3 +265,20 @@ div.localization
+  padding: 2pt;
+}
+
+.qa_details details summary {
```
I think having a :hover style for the summary which changes the mouse to pointer would be a good usability addition.

bitschmidty replied:
The hover style looks good!

harding commented:
@bitschmidty 

> Suggest mouse pointer hover style (noted inline comment)

Good idea!  Implemented.

> Doesn’t seem we have an anchoring capabilities for the qa blocks or individual questions/answers.

Even better idea!  Implemented.  Anchors are attached to the entire details box, using the question text to create the slug.  We don't have an easy glyph to attach the anchors like the bullet in #369, so I added a link icon to end of the answer field, which I think is ok---but I'm happy to try something else.  Unfortunately, there's no way to trigger the opening of the details box using pure CSS so anyone following a link to a particular question will still have to click on the box (or we'll have to use JS, yuck).

I left the email version free of anchors or links, since I wasn't sure whether that'd be useful.  I can easily add them---let me know.

> My flow would still be to copy/paste the newsletter content from the live production site, since all of the links are correct there. [...] Or is there a better approach for me to take here? (building prod locally w/ correct domain name, etc)

#389 should fix the domain issue.

---

If everything is looking basically ok, I'll convert the remaining lists to use this function.

jnewbery commented on _posts/en/newsletters/2020-04-08-newsletter.md:
```diff
@@ -7,7 +7,7 @@ type: newsletter
 lang: en
 ---
-This week's newsletter describes work on simplified ECDSA adaptor
+Foo This week's newsletter describes work on simplified ECDSA adaptor
```
Barrrr

harding replied:
Whoops.  Fixed.  Thanks!

jonatack commented on _includes/functions/details-list.md:
```diff
@@ -0,0 +1,15 @@
+<div class="{{include.class | default: 'qa_details'}} {{jekyll.environment}}" markdown="1">
+{% for i in (0..9) %}
```
9 questions/answers seems a reasonable limit; should it be made explicit somewhere?

jonatack commented on _posts/en/newsletters/2020-04-08-newsletter.md:
```diff
@@ -79,25 +79,29 @@ to obtain the transaction as fast as possible from another peer.
+
+  q4="Describe one issue with removing `mapRelay`?"
+  a4="It could cause requested transactions in honest situations to more often be
+      `notfound` with delays of up to 2 minutes, hurting propagation."
```
Markdown works here now? Nice!

jonatack replied:
Oh, `markdown="1"` seem to do it :+1:

jonatack replied:
Oh right, `markdown="1"` seems to do it :+1:

harding commented:
@jonatack 

> would it be feasible for it to convert markdown code chars to html code tags?

I should've mentioned it in my OP, but this will indeed make that conversion, providing full markdown conversion in the context of the evaluated page (so, e.g., we can put reference-style link definitions at the bottom of the page like normal).  For the sake of a demo, I quickly converted the first section of your section from last week to the new format (changing the code tags to backtics) so you can see that it works.  See: https://deploy-preview-387--bitcoinops.netlify.com/en/newsletters/2020/04/08/#why-could-retrying-notfound-more-quickly-be-helpful

> 9 questions/answers seems a reasonable limit; should it be made explicit somewhere?

Good idea.  I'll see if I can get it to fail to build if anyone tries more than 9.

Thanks for the review!

jnewbery commented:
Concept ACK. I think this can be updated to use simplified styling for JEKYLL_ENV=email now that #389 is merged.

harding commented:
All feedback should be addressed (thanks!).

jnewbery commented on _includes/functions/details-list.md:
```diff
@@ -0,0 +1,25 @@
+   - <i markdown="1">{{include.[q]}}</i><br>{{include.[a]}}
+  {% else %}
+   - <details id="{{qslug}}" markdown="1"><summary><span markdown="1">{{include.[q]}}</span></summary>
+     {{include.[a]}}&nbsp;<a href="#{{qslug}}">🔗</a>
```
I suggest we drop the link icon here:

- I think the most intuitive interpretation is that clicking on this will link to the discussion or source, not this section of the newsletter
- it's unlikely we (or anyone else) will need to link to these often. If we do, we can just inspect the html and grab the id from there.

harding commented:
Dropped link icon.  Also tested in an RSS feed reader (not great, but ok I think):

![2020-04-30-06_43_52_892822698](https://user-images.githubusercontent.com/61096/80701822-03becd00-8aae-11ea-9b6a-be85815ce61b.png)
